### PR TITLE
feat: Integrate feemarket + reworked ante, post handler decorator distinction

### PR DESCRIPTION
**Pull request by vishal-kanna**
closes: #777

### CODE REVIEW

File: ante/fee_test.go
Comment (Eengineer1) [inline ante/fee_test.go:1]: Unit tests _must_ be eventually updated as well; either in parallel or finalising the implementation.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
Let's adjust the antehandler to route transactions based on the aforementioned remarks + restore the indicative unit tests.

Posthandler looks closer to completion.

File: ante/fee.go
[the remark this answers is not in the document]
Comment (vishal-kanna) [reply ante/fee.go:1]: Ok @Eengineer1

**Review by Eengineer1 — COMMENTED** [review]
@vishal-kanna Could you please remove comments + temporarily disable upgrade tests preparing the PR for a final review?

File: tests/integration/cli/config.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ const (
 	KeyringBackend = "test"
 	OutputFormat   = "json"
 	Gas            = "auto"
-	GasAdjustment  = "1.8"
-	GasPrices      = "50ncheq"
+	GasAdjustment  = "3.8"
+	GasPrices      = "90ncheq"
```
Comment (Eengineer1) [inline tests/integration/cli/config.go:10]: Let's try to adjust back to the previous values and / or slightly increase if necessary, but not such unrealistic quantities.

e.g. Try with:

```suggestion
	GasAdjustment  = "1.8"
	GasPrices      = "50ncheq"
```

or 

```suggestion
	GasAdjustment  = "2.0"
	GasPrices      = "60ncheq"
```

File: tests/integration/helpers/fee.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ func GenerateFees(amount string) []string {
 	return []string{
 		"--fees", amount,
 		"--gas", "auto",
-		"--gas-adjustment", "2.5",
+		"--gas-adjustment", "3.5",
```
Comment (Eengineer1) [inline tests/integration/helpers/fee.go:11]: Similarly, let's settle with a singular all-purpose realistic value as suggested above.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]

File: tests/integration/cli/config.go (hunk 1 above)
Comment (vishal-kanna) [reply tests/integration/cli/config.go:10]: Ok @Eengineer1

File: app/app.go [hunk 3]
```diff
@@ -493,6 +502,9 @@ func New(
 		),
 	)
 
+	app.FeeMarketKeeper = feemarketkeeper.NewKeeper(appCodec, keys[feemarkettypes.StoreKey], app.AccountKeeper, &feemarkettypes.TestDenomResolver{}, authtypes.NewModuleAddress(govtypes.ModuleName).String())
+	app.FeeMarketKeeper.SetDenomResolver(&feemarkettypes.TestDenomResolver{}) // TODO
```
Comment (Eengineer1) [inline app/app.go:506]: Looks like this is still pending.

Could you address + resolve?

File: app/app.go
Comment (Eengineer1) [inline app/app.go:1]: Don't we need to update the upgrade store loader respectively to include the newly introduced modules?

Looks like that's missing.

If unclear still, please follow the recommended upgrade store logic described [here](https://docs.cosmos.network/v0.47/learn/advanced/upgrade#adding-new-modules-during-upgrades).

File: go.mod [hunk 4]
```diff
@@ -1,76 +1,76 @@
 module github.com/cheqd/cheqd-node
 
-go 1.21
-
-toolchain go1.21.0
+go 1.22.6
```
Comment (Eengineer1) [inline go.mod:3]: Let's assess whether any Go version bumps are necessary within GH actions.

File: tests/integration/cli/config.go
Comment (Eengineer1) [inline tests/integration/cli/config.go:1]: The values look more consistent with the live network respectively.

Nice work.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
The core work looks good.

Please address the remarks before merging the work into develop + synchronising across the various dependent PRs, such as the fee-abstraction integration.

File: ante/fee_test.go [hunk 5]
```diff
@@ -91,6 +130,14 @@ var _ = Describe("Fee tests on CheckTx", func() {
 		Expect(err).NotTo(BeNil(), "Decorator should have errored on too low fee for local gasPrice")
 
 		// antehandler should not error since we do not check minGasPrice in simulation mode
+		params, err = s.app.FeeMarketKeeper.GetParams(s.ctx)
+		Expect(err).To(BeNil())
+
+		// Set high gas price in feemarket
+		params.Enabled = false
```
[the remark this answers is not in the document]
Comment (vishal-kanna) [reply ante/fee_test.go:137]: In anteHandler first feemarket params are checked first .If we enable it to true below case doesn't run(i.e in simulation mode).In order to run in simulation mode we are disabling it.

[the remark this answers is not in the document]
Comment (Eengineer1) [reply ante/fee_test.go:137]: I've figured as such, but still wanted to cross-check + validate that with you.

Thanks!

File: app/app.go (hunk 3 above)
Comment (vishal-kanna) [reply app/app.go:507]: This is handled in the feebs PR.

**Review by Eengineer1 — APPROVED** [review]
The standalone work looks well polished.

Approved.

**Review by Eengineer1 — APPROVED** [review]
Approved with increased gas adjustment for consistency purposes.

**Pull request merged**